### ISSUE TITLE
Performance optimization on ProductLoadAllPaged procedure

### ISSUE THREAD

**Issue by skoshelev** [nopSolutions/nopCommerce#2897]
nopCommerce version: 4.10

Sometimes executing of this procedure takes too much time. We have to optimize it. 
First of all let's change "left join" to "inner join" in the following code:

```
IF @CategoryIdsCount > 0
BEGIN
    SET @sql = @sql + '
    LEFT JOIN Product_Category_Mapping pcm with (NOLOCK)
        ON p.Id = pcm.ProductId'
END

IF @ManufacturerId > 0
BEGIN
    SET @sql = @sql + '
    LEFT JOIN Product_Manufacturer_Mapping pmm with (NOLOCK)
        ON p.Id = pmm.ProductId'
END
```

Then let's investigate other ways to boost performance

**Comment by ivanslater** [comment]
Tested ok... little performance improvement was added! no bugs or problems detected in my tests.

**Comment by skoshelev** [comment]
Closed #2897
